sockeye/arguments.py

```diff
@@ -529,9 +548,17 @@ def add_training_args(params):
                               help="Dropout probability for ConvolutionalEmbeddingEncoder. Default: %(default)s.")
 
     train_params.add_argument('--optimizer',
-                              default='adam',
-                              choices=['adam', 'sgd', 'rmsprop'],
+                              default=C.OPTIMIZER_ADAM,
+                              choices=C.OPTIMIZERS,
                               help='SGD update rule. Default: %(default)s.')
+    train_params.add_argument('--optimizer-params',
+                              type=json_dict(),
+                              default=None,
+                              help='Additional optimizer params as JSON dictionary, should only be used for testing.')
```

Comment:
It turns out this is actually a problematic format since it uses both quoting and spacing, making it less than ideal for scripts.  I'm going to work out a new format that we can use as the Plan A for passing args to optimizers.